### ISSUE TITLE
Planner bug: Weight string added to the wrong expression

### ISSUE THREAD

**Issue by systay** [vitessio/vitess#16326]
For queries where we are missing type information, we sometimes add `weight_string()` calls as a way to get an easily comparable value.

When planning `min`/`max` on columns where we are missing type info, we need to wrap the values in `weight_string()`. 

If we have been able to push down aggregation, we want the weight_string of the aggregate function. If we haven't been able to push it down, we need the weight_string of the argument to the function.

For this query:

```sql
select min(foo) from user
```

We are producing the plan:

```json
{
  "QueryType": "SELECT",
  "Original": "select min(foo) from user",
  "Instructions": {
    "OperatorType": "Aggregate",
    "Variant": "Scalar",
    "Aggregates": "min(0|1) AS min(foo)",
    "ResultColumns": 1,
    "Inputs": [
      {
        "OperatorType": "Route",
        "Variant": "Scatter",
        "Keyspace": {
          "Name": "user",
          "Sharded": true
        },
        "FieldQuery": "select min(foo), weight_string(foo) from `user` where 1 != 1 group by weight_string(foo)",
        "Query": "select min(foo), weight_string(foo) from `user` group by weight_string(foo)",
        "Table": "`user`"
      }
    ]
  },
  "TablesUsed": [
    "user.user"
  ]
}
```

As we can see, the MySQL query is selecting `weight_string(foo) ` instead of `weight_string(min(foo)) `

**Comment by GrahamCampbell** [comment]
I believe this issue was fixed, at least in v21, by #16325, but was not auto-closed because of a double hash?

**Comment by systay** [comment]
@GrahamCampbell you are right. thanks!
